sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubBufferedProducerClientBuilder.java

```diff
@@ -353,14 +367,40 @@ public EventHubBufferedProducerClientBuilder transportType(AmqpTransportType tra
         return this;
     }
 
-
     /**
      * Builds a new instance of the async buffered producer client.
      *
      * @return A new instance of {@link EventHubBufferedProducerAsyncClient}.
+     *
+     * @throws NullPointerException if {@link #onSendBatchSucceeded(Consumer)}, {@link
+     *     #onSendBatchFailed(Consumer)}, or {@link #maxWaitTime(Duration)} are null.
+     * @throws IllegalArgumentException if {@link #maxConcurrentSends(int)}, {@link
+     *     #maxConcurrentSendsPerPartition(int)}, or {@link #maxEventBufferLengthPerPartition(int)} are less than 1.
      */
     public EventHubBufferedProducerAsyncClient buildAsyncClient() {
-        return new EventHubBufferedProducerAsyncClient(builder, clientOptions);
+
+        if (Objects.isNull(clientOptions.getSendSucceededContext())) {
+            throw LOGGER.logExceptionAsError(new NullPointerException("'onSendBatchSucceeded' cannot be null."));
+        } else if (Objects.isNull(clientOptions.getSendFailedContext())) {
```

Comment:
remove all the `else` keywords? It looks like they are `and` relationship not `or-else` relationship?

Reply:
Sure. I was trying to save white space. :) But fixed.